### PR TITLE
rustfmt: format comments

### PR DESCRIPTION
**Pull request by buffalojoec**
This PR adds comment formatting configurations to the repository's `rustfmt.toml` file, and the associated changes from `cargo +nightly fmt --all`.

Comment width 80.

### CODE REVIEW

**Comment by buffalojoec** [comment]
Current dependencies on/for this PR:
* master
  * **PR #5741** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-program-library/5741" target="_blank">Graphite</a> 
    * **PR #5742** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-program-library/5742" target="_blank">Graphite</a> 
      * **PR #5743** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-program-library/5743" target="_blank">Graphite</a> 
        * **PR #5744** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-program-library/5744" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/solana-labs/solana-program-library/5744?utm_source=stack-comment).

File: governance/program/src/instruction.rs [hunk 1]
```diff
@@ -94,23 +103,30 @@ pub enum GovernanceInstruction {
...
     },
 
-    /// Withdraws governing tokens (Community or Council) from Governance Realm and downgrades your voter weight within the Realm
-    /// Note: It's only possible to withdraw tokens if the Voter doesn't have any outstanding active votes
-    /// If there are any outstanding votes then they must be relinquished before tokens could be withdrawn
+    /// Withdraws governing tokens (Community or Council) from Governance Realm
+    /// and downgrades your voter weight within the Realm Note: It's only
+    /// possible to withdraw tokens if the Voter doesn't have any outstanding
+    /// active votes If there are any outstanding votes then they must be
+    /// relinquished before tokens could be withdrawn
```
Comment (joncinque) [inline governance/program/src/instruction.rs:110]: Need some more newlines, not exactly like this, but you get the idea
```suggestion
    /// and downgrades your voter weight within the Realm.
    /// Note: It's only
    /// possible to withdraw tokens if the Voter doesn't have any outstanding
    /// active votes.
    /// If there are any outstanding votes then they must be
    /// relinquished before tokens could be withdrawn
```

File: governance/program/src/instruction.rs [hunk 2]
```diff
@@ -94,23 +103,30 @@ pub enum GovernanceInstruction {
...
     ///  6. `[]` RealmConfig account. PDA seeds: ['realm-config', realm]
     WithdrawGoverningTokens {},
 
-    /// Sets Governance Delegate for the given Realm and Governing Token Mint (Community or Council)
-    /// The Delegate would have voting rights and could vote on behalf of the Governing Token Owner
-    /// The Delegate would also be able to create Proposals on behalf of the Governing Token Owner
-    /// Note: This doesn't take voting rights from the Token Owner who still can vote and change governance_delegate
+    /// Sets Governance Delegate for the given Realm and Governing Token Mint
+    /// (Community or Council) The Delegate would have voting rights and
+    /// could vote on behalf of the Governing Token Owner The Delegate would
```
Comment (joncinque) [inline governance/program/src/instruction.rs:126]:
```suggestion
    /// (Community or Council). The Delegate would have voting rights and
    /// could vote on behalf of the Governing Token Owner. The Delegate would
```

File: governance/program/src/instruction.rs [hunk 3]
```diff
@@ -160,27 +184,34 @@ pub enum GovernanceInstruction {
...
     ///   9. `[]` Optional Voter Weight Record
-    ///   10.`[writable]` Optional ProposalDeposit account. PDA seeds: ['proposal-deposit', proposal, deposit payer]
-    ///       Proposal deposit is required when there are more active proposals than the configured deposit exempt amount
-    ///       The deposit is paid by the Payer of the transaction and can be reclaimed using RefundProposalDeposit once the Proposal is no longer active
+    ///   10.`[writable]` Optional ProposalDeposit account. PDA seeds:
+    /// ['proposal-deposit', proposal, deposit payer]       Proposal deposit
+    /// is required when there are more active proposals than the configured
+    /// deposit exempt amount       The deposit is paid by the Payer of the
+    /// transaction and can be reclaimed using RefundProposalDeposit once the
+    /// Proposal is no longer active
```
Comment (joncinque) [inline governance/program/src/instruction.rs:214]: These should be indented

File: governance/program/src/instruction.rs [hunk 4]
```diff
@@ -273,40 +314,51 @@ pub enum GovernanceInstruction {
...
+    ///      Delegate)
+    ///   6. `[writable]` Proposal VoteRecord account. PDA seeds:
+    ///      ['governance',proposal,token_owner_record]
+    ///   7. `[]` The Governing Token Mint which is used to cast the vote
+    ///      (vote_governing_token_mint) The voting token mint is the
+    ///      governing_token_mint of the Proposal for Approve, Deny and Abstain
+    ///      votes For Veto vote the voting token mint is the mint of the
+    ///      opposite voting population Council mint to veto Community proposals
+    ///      and Community mint to veto Council proposals Note: In the current
+    ///      version only Council veto is supported
```
Comment (joncinque) [inline governance/program/src/instruction.rs:361]: Need some periods here
```suggestion
    ///   7. `[]` The Governing Token Mint which is used to cast the vote
    ///      (vote_governing_token_mint). The voting token mint is the
    ///      governing_token_mint of the Proposal for Approve, Deny and Abstain
    ///      votes. For Veto vote the voting token mint is the mint of the
    ///      opposite voting population. Council mint to veto Community proposals
    ///      and Community mint to veto Council proposals.
    ///      Note: In the current
    ///      version only Council veto is supported
```

File: governance/program/src/instruction.rs [hunk 5]
```diff
@@ -329,27 +382,36 @@ pub enum GovernanceInstruction {
...
-    ///  Relinquish Vote removes voter weight from a Proposal and removes it from voter's active votes
-    ///  If the Proposal is still being voted on then the voter's weight won't count towards the vote outcome
-    ///  If the Proposal is already in decided state then the instruction has no impact on the Proposal
-    ///  and only allows voters to prune their outstanding votes in case they wanted to withdraw Governing tokens from the Realm
+    ///  Relinquish Vote removes voter weight from a Proposal and removes it
+    /// from voter's active votes  If the Proposal is still being voted on
+    /// then the voter's weight won't count towards the vote outcome  If the
+    /// Proposal is already in decided state then the instruction has no impact
+    /// on the Proposal  and only allows voters to prune their outstanding
+    /// votes in case they wanted to withdraw Governing tokens from the Realm
```
Comment (joncinque) [inline governance/program/src/instruction.rs:390]: This one needs periods

File: governance/program/src/instruction.rs [hunk 6]
```diff
@@ -377,19 +442,24 @@ pub enum GovernanceInstruction {
...
 
         #[allow(dead_code)]
-        /// Indicates whether Mint's authorities (MintTokens, FreezeAccount) should be transferred to the Governance PDA
-        /// If it's set to false then it can be done at a later time
-        /// However the instruction would validate the current mint authority signed the transaction nonetheless
+        /// Indicates whether Mint's authorities (MintTokens, FreezeAccount)
+        /// should be transferred to the Governance PDA If it's set to
+        /// false then it can be done at a later time However the
+        /// instruction would validate the current mint authority signed the
+        /// transaction nonetheless
```
Comment (joncinque) [inline governance/program/src/instruction.rs:449]: This needs some periods

File: governance/program/src/instruction.rs [hunk 7]
```diff
@@ -420,21 +492,25 @@ pub enum GovernanceInstruction {
     },
 
     /// Flags a transaction and its parent Proposal with error status
-    /// It can be used by Proposal owner in case the transaction is permanently broken and can't be executed
-    /// Note: This instruction is a workaround because currently it's not possible to catch errors from CPI calls
-    ///       and the Governance program has no way to know when instruction failed and flag it automatically
+    /// It can be used by Proposal owner in case the transaction is permanently
+    /// broken and can't be executed Note: This instruction is a workaround
```
Comment (joncinque) [inline governance/program/src/instruction.rs:496]: newline probably
```suggestion
    /// broken and can't be executed.
    /// Note: This instruction is a workaround
```

File: governance/program/src/instruction.rs
Comment (joncinque) [inline governance/program/src/instruction.rs:1]: This file is rough in general

File: governance/program/src/instruction.rs [hunk 8]
```diff
@@ -444,66 +520,81 @@ pub enum GovernanceInstruction {
...
-    ///       Note: In the current version it's only possible to remove council mint (set it to None)
-    ///       After setting council to None it won't be possible to withdraw the tokens from the Realm any longer
-    ///       If that's required then it must be done before executing this instruction
-    ///   3. `[writable]` Council Token Holding account - optional unless council is used. PDA seeds: ['governance',realm,council_mint]
-    ///       The account will be created with the Realm PDA as its owner
+    ///   2. `[]` Council Token Mint - optional Note: In the current version
+    ///      it's only possible to remove council mint (set it to None) After
+    ///      setting council to None it won't be possible to withdraw the tokens
+    ///      from the Realm any longer If that's required then it must be done
+    ///      before executing this instruction
```
Comment (joncinque) [inline governance/program/src/instruction.rs:527]: this one is pretty rough

File: governance/program/src/instruction.rs [hunk 9]
```diff
@@ -444,66 +520,81 @@ pub enum GovernanceInstruction {
...
+    ///     governance]
     ///  2. `[signer]` Payer
     ///  3. `[]` System
     CreateNativeTreasury,
 
-    /// Revokes (burns) membership governing tokens for the given TokenOwnerRecord and hence takes away governance power from the TokenOwner
-    /// Note: If there are active votes for the TokenOwner then the vote weights won't be updated automatically
+    /// Revokes (burns) membership governing tokens for the given
+    /// TokenOwnerRecord and hence takes away governance power from the
+    /// TokenOwner Note: If there are active votes for the TokenOwner then
```
Comment (joncinque) [inline governance/program/src/instruction.rs:584]:
```suggestion
    /// TokenOwner. Note: If there are active votes for the TokenOwner then
```

File: single-pool/program/src/instruction.rs [hunk 10]
```diff
@@ -24,9 +24,10 @@ use {
 #[repr(C)]
 #[derive(Clone, Debug, PartialEq, BorshSerialize, BorshDeserialize)]
 pub enum SinglePoolInstruction {
-    ///   Initialize the mint and stake account for a new single-validator stake pool.
-    ///   The pool stake account must contain the rent-exempt minimum plus the minimum delegation.
-    ///   No tokens will be minted: to deposit more, use `Deposit` after `InitializeStake`.
+    ///   Initialize the mint and stake account for a new single-validator stake
+    /// pool.   The pool stake account must contain the rent-exempt minimum
+    /// plus the minimum delegation.   No tokens will be minted: to deposit
+    /// more, use `Deposit` after `InitializeStake`.
```
Comment (joncinque) [inline single-pool/program/src/instruction.rs:30]: This can be cleaned up

File: single-pool/program/src/instruction.rs [hunk 11]
```diff
@@ -43,8 +44,9 @@ pub enum SinglePoolInstruction {
     ///  12. `[]` Stake program
     InitializePool,
 
-    ///   Restake the pool stake account if it was deactivated. This can happen through the
-    ///   stake program's `DeactivateDelinquent` instruction, or during a cluster restart.
+    ///   Restake the pool stake account if it was deactivated. This can happen
+    /// through the   stake program's `DeactivateDelinquent` instruction, or
+    /// during a cluster restart.
```
Comment (joncinque) [inline single-pool/program/src/instruction.rs:49]: This too, can be cleaned up

File: single-pool/program/src/instruction.rs [hunk 12]
```diff
@@ -56,8 +58,9 @@ pub enum SinglePoolInstruction {
     ///   7. `[]` Stake program
     ReactivatePoolStake,
 
-    ///   Deposit stake into the pool.  The output is a "pool" token representing fractional
-    ///   ownership of the pool stake. Inputs are converted to the current ratio.
+    ///   Deposit stake into the pool.  The output is a "pool" token
+    /// representing fractional   ownership of the pool stake. Inputs are
+    /// converted to the current ratio.
```
Comment (joncinque) [inline single-pool/program/src/instruction.rs:63]: This three, can be cleaned up

File: single-pool/program/src/instruction.rs [hunk 13]
```diff
@@ -92,10 +95,10 @@ pub enum SinglePoolInstruction {
...
     },
 
-    ///   Create token metadata for the stake-pool token in the metaplex-token program.
-    ///   Step three of the permissionless three-stage initialization flow.
-    ///   Note this instruction is not necessary for the pool to operate, to ensure we cannot
-    ///   be broken by upstream.
+    ///   Create token metadata for the stake-pool token in the metaplex-token
+    /// program.   Step three of the permissionless three-stage
+    /// initialization flow.   Note this instruction is not necessary for
+    /// the pool to operate, to ensure we cannot   be broken by upstream.
```
Comment (joncinque) [inline single-pool/program/src/instruction.rs:101]: This can be cleaned up

File: stake-pool/program/src/big_vec.rs [hunk 14]
```diff
@@ -70,7 +70,8 @@ impl<'data> BigVec<'data> {
             let gap = removals_found * mem::size_of::<T>();
             // In case the compute budget is ever bumped up, allowing us
             // to use this safe code instead:
-            //self.data.copy_within(dst_start_index + gap..data_end_index, dst_start_index);
+            //self.data.copy_within(dst_start_index + gap..data_end_index,
+            // dst_start_index);
```
Comment (joncinque) [inline stake-pool/program/src/big_vec.rs:74]: bummer

File: token/transfer-hook/example/src/processor.rs [hunk 15]
```diff
@@ -73,7 +73,9 @@ pub fn process_execute(
     Ok(())
 }
 
-/// Processes a [InitializeExtraAccountMetaList](enum.TransferHookInstruction.html) instruction.
+/// Processes a
+/// [InitializeExtraAccountMetaList](enum.TransferHookInstruction.html)
+/// instruction.
```
Comment (joncinque) [inline token/transfer-hook/example/src/processor.rs:78]: I don't even remember if these links work, but we can leave as is

File: token/program/src/lib.rs [hunk 16]
```diff
@@ -53,12 +56,13 @@ pub fn amount_to_ui_amount_string_trimmed(amount: u64, decimals: u8) -> String {
...
-/// Try to convert a UI representation of a token amount to its raw amount using the given decimals
-/// field
+/// Try to convert a UI representation of a token amount to its raw amount using
+/// the given decimals field
 pub fn try_ui_amount_into_amount(ui_amount: String, decimals: u8) -> Result<u64, ProgramError> {
     let decimals = decimals as usize;
     let mut parts = ui_amount.split('.');
-    let mut amount_str = parts.next().unwrap().to_string(); // splitting a string, even an empty one, will always yield an iterator of at least length == 1
+    let mut amount_str = parts.next().unwrap().to_string(); // splitting a string, even an empty one, will always yield an iterator of at
+                                                            // least length == 1
```
Comment (joncinque) [inline token/program/src/lib.rs:65]: we can just put the comment above here

File: token/program-2022/src/lib.rs [hunk 17]
```diff
@@ -66,12 +69,13 @@ pub fn amount_to_ui_amount_string_trimmed(amount: u64, decimals: u8) -> String {
...
-/// Try to convert a UI representation of a token amount to its raw amount using the given decimals
-/// field
+/// Try to convert a UI representation of a token amount to its raw amount using
+/// the given decimals field
 pub fn try_ui_amount_into_amount(ui_amount: String, decimals: u8) -> Result<u64, ProgramError> {
     let decimals = decimals as usize;
     let mut parts = ui_amount.split('.');
-    let mut amount_str = parts.next().unwrap().to_string(); // splitting a string, even an empty one, will always yield an iterator of at least length == 1
+    let mut amount_str = parts.next().unwrap().to_string(); // splitting a string, even an empty one, will always yield an iterator of at
+                                                            // least length == 1
```
Comment (joncinque) [inline token/program-2022/src/lib.rs:78]: can move this comment above

File: token/program-2022/src/extension/mod.rs [hunk 18]
```diff
@@ -272,8 +273,9 @@ fn check_account_type<S: BaseState>(account_type: AccountType) -> Result<(), Pro
 ///                          ^     ^       ^     ^
 ///                     acct type  extension length data...
 ///
-/// Mint: 82 bytes... + 83 bytes of other extension data + [2, 0, 3, 0, 100, ....]
-///                                                         ^ data in extension just happens to look like this
+/// Mint: 82 bytes... + 83 bytes of other extension data + [2, 0, 3, 0, 100,
+/// ....]                                                         ^ data in
+/// extension just happens to look like this
```
Comment (joncinque) [inline token/program-2022/src/extension/mod.rs:278]: Eh, this is not good. might need to do something like
```suggestion
/// Mint: 82 bytes... + 83 bytes of other extension data + 
/// [2, 0, 3, 0, 100, ....]
///       ^ data in extension just happens to look like this
```

File: token/program-2022/src/extension/interest_bearing_mint/mod.rs [hunk 19]
```diff
@@ -111,12 +111,13 @@ impl InterestBearingConfig {
         if amount > (u64::MAX as f64) || amount < (u64::MIN as f64) || amount.is_nan() {
             Err(ProgramError::InvalidArgument)
         } else {
-            Ok(amount.round() as u64) // this is important, if you round earlier, you'll get wrong "inf" answers
+            Ok(amount.round() as u64) // this is important, if you round
+                                      // earlier, you'll get wrong "inf" answers
```
Comment (joncinque) [inline token/program-2022/src/extension/interest_bearing_mint/mod.rs:115]: can move the comment above

File: stake-pool/program/src/instruction.rs [hunk 20]
```diff
@@ -80,7 +82,8 @@ pub enum StakePoolInstruction {
     ///   list of managed validators.
     ///
     ///   The stake account will have the rent-exempt amount plus
-    ///   `max(crate::MINIMUM_ACTIVE_STAKE, solana_program::stake::tools::get_minimum_delegation())`.
+    ///   `max(crate::MINIMUM_ACTIVE_STAKE,
+    /// solana_program::stake::tools::get_minimum_delegation())`.
```
Comment (joncinque) [inline stake-pool/program/src/instruction.rs:86]:
```suggestion
    ///     solana_program::stake::tools::get_minimum_delegation())`.
```

File: stake-pool/program/src/instruction.rs [hunk 21]
```diff
@@ -104,8 +107,9 @@ pub enum StakePoolInstruction {
     ///   (Staker only) Removes validator from the pool, deactivating its stake
     ///
     ///   Only succeeds if the validator stake account has the minimum of
-    ///   `max(crate::MINIMUM_ACTIVE_STAKE, solana_program::stake::tools::get_minimum_delegation())`.
-    ///   plus the rent-exempt amount.
+    ///   `max(crate::MINIMUM_ACTIVE_STAKE,
+    /// solana_program::stake::tools::get_minimum_delegation())`.   plus the
```
Comment (joncinque) [inline stake-pool/program/src/instruction.rs:111]:
```suggestion
    ///    solana_program::stake::tools::get_minimum_delegation())`.   plus the
```

File: stake-pool/program/src/instruction.rs [hunk 22]
```diff
@@ -250,19 +262,24 @@ pub enum StakePoolInstruction {
     ///   1. `[w]` Validator stake list storage account
     CleanupRemovedValidatorEntries,
 
-    ///   Deposit some stake into the pool.  The output is a "pool" token representing ownership
-    ///   into the pool. Inputs are converted to the current ratio.
+    ///   Deposit some stake into the pool.  The output is a "pool" token
+    /// representing ownership   into the pool. Inputs are converted to the
+    /// current ratio.
```
Comment (joncinque) [inline stake-pool/program/src/instruction.rs:267]: spacing weird here

File: stake-pool/program/src/instruction.rs [hunk 23]
```diff
@@ -272,10 +289,11 @@ pub enum StakePoolInstruction {
 
     ///   Withdraw the token from the pool at the current ratio.
     ///
-    ///   Succeeds if the stake account has enough SOL to cover the desired amount
-    ///   of pool tokens, and if the withdrawal keeps the total staked amount
-    ///   above the minimum of rent-exempt amount +
-    ///   `max(crate::MINIMUM_ACTIVE_STAKE, solana_program::stake::tools::get_minimum_delegation())`.
+    ///   Succeeds if the stake account has enough SOL to cover the desired
+    /// amount   of pool tokens, and if the withdrawal keeps the total
+    /// staked amount   above the minimum of rent-exempt amount +
```
Comment (joncinque) [inline stake-pool/program/src/instruction.rs:294]: spacing weird here

File: stake-pool/program/src/instruction.rs [hunk 24]
```diff
@@ -329,8 +347,9 @@ pub enum StakePoolInstruction {
     ///  2. '[]` New staker pubkey
     SetStaker,
 
-    ///   Deposit SOL directly into the pool's reserve account. The output is a "pool" token
-    ///   representing ownership into the pool. Inputs are converted to the current ratio.
+    ///   Deposit SOL directly into the pool's reserve account. The output is a
+    /// "pool" token   representing ownership into the pool. Inputs are
+    /// converted to the current ratio.
```
Comment (joncinque) [inline stake-pool/program/src/instruction.rs:352]: weird spacing

File: stake-pool/program/src/instruction.rs [hunk 25]
```diff
@@ -575,20 +610,24 @@ pub enum StakePoolInstruction {
         destination_transient_stake_seed: u64,
     },
 
-    ///   Deposit some stake into the pool, with a specified slippage constraint.
-    ///   The output is a "pool" token representing ownership into the pool.
-    ///   Inputs are converted at the current ratio.
+    ///   Deposit some stake into the pool, with a specified slippage
+    /// constraint.   The output is a "pool" token representing ownership
+    /// into the pool.   Inputs are converted at the current ratio.
```
Comment (joncinque) [inline stake-pool/program/src/instruction.rs:615]: weird spacing

File: stake-pool/program/src/instruction.rs [hunk 26]
```diff
@@ -602,10 +641,11 @@ pub enum StakePoolInstruction {
...
     ///   minimum expected output lamport amount.
     ///
-    ///   Succeeds if the stake account has enough SOL to cover the desired amount
-    ///   of pool tokens, and if the withdrawal keeps the total staked amount
-    ///   above the minimum of rent-exempt amount +
-    ///   `max(crate::MINIMUM_ACTIVE_STAKE, solana_program::stake::tools::get_minimum_delegation())`.
+    ///   Succeeds if the stake account has enough SOL to cover the desired
+    /// amount   of pool tokens, and if the withdrawal keeps the total
+    /// staked amount   above the minimum of rent-exempt amount +
+    ///   `max(crate::MINIMUM_ACTIVE_STAKE,
```
Comment (joncinque) [inline stake-pool/program/src/instruction.rs:647]: weird spacing

File: stake-pool/program/src/instruction.rs [hunk 27]
```diff
@@ -629,8 +669,9 @@ pub enum StakePoolInstruction {
     },
 
     ///   Deposit SOL directly into the pool's reserve account, with a specified
-    ///   slippage constraint. The output is a "pool" token representing ownership
-    ///   into the pool. Inputs are converted at the current ratio.
+    ///   slippage constraint. The output is a "pool" token representing
+    /// ownership   into the pool. Inputs are converted at the current
+    /// ratio.
```
Comment (joncinque) [inline stake-pool/program/src/instruction.rs:674]: weird spacing

File: token-lending/program/tests/modify_reserve_config.rs [hunk 28]
```diff
@@ -366,7 +367,9 @@ async fn correct_owner_providing_wrong_lending_market_fails() {
             new_config,
             sol_test_reserve.pubkey,
             other_lending_market.pubkey,
-            lending_market.owner.pubkey(), //lending_market.owner == other_lending_market.owner, defined by `add_lending_market`
+            lending_market.owner.pubkey(), /* lending_market.owner ==
+                                            * other_lending_market.owner, defined by
+                                            * `add_lending_market` */
```
Comment (joncinque) [inline token-lending/program/tests/modify_reserve_config.rs:372]: can move the comment above

File: token-swap/program/src/curve/constant_product.rs [hunk 29]
```diff
@@ -390,15 +391,20 @@ mod tests {
...
             (10, 20_000 - 9, 30_000, 10, 14), // spot: 10 * 3 / 2.0001 = 14.999
             (10, 20_000 - 10, 30_000, 10, 15), // spot: 10 * 3 / 2.0000 = 15
-            (100, 60_000, 30_000, 99, 49), // spot: 100 * 3 / 6.001 = 49.99 (source can be 99 to get 49 dest.)
-            (99, 60_000, 30_000, 99, 49),  // spot: 99 * 3 / 6.001 = 49.49
-            (98, 60_000, 30_000, 97, 48), // spot: 98 * 3 / 6.001 = 48.99 (source can be 97 to get 48 dest.)
+            (100, 60_000, 30_000, 99, 49),    /* spot: 100 * 3 / 6.001 = 49.99 (source can be 99
+                                               * to get 49 dest.) */
+            (99, 60_000, 30_000, 99, 49), // spot: 99 * 3 / 6.001 = 49.49
+            (98, 60_000, 30_000, 97, 48), /* spot: 98 * 3 / 6.001 = 48.99 (source can be 97 to
+                                           * get 48 dest.) */
```
Comment (joncinque) [inline token-swap/program/src/curve/constant_product.rs:407]: these kinda stink, might be better to move the comment above

File: token-swap/program/src/instruction.rs [hunk 30]
```diff
@@ -141,23 +150,25 @@ pub enum SwapInstruction {
...
     ///   12. `[]` Token B program id
     ///   13. `[]` Pool Token program id
     DepositAllTokenTypes(DepositAllTokenTypes),
 
-    ///   Withdraw both types of tokens from the pool at the current ratio, given
-    ///   pool tokens.  The pool tokens are burned in exchange for an equivalent
-    ///   amount of token A and B.
+    ///   Withdraw both types of tokens from the pool at the current ratio,
+    /// given   pool tokens.  The pool tokens are burned in exchange for an
+    /// equivalent   amount of token A and B.
```
Comment (joncinque) [inline token-swap/program/src/instruction.rs:164]: spacing off here

File: token-swap/program/src/instruction.rs [hunk 31]
```diff
@@ -170,18 +181,20 @@ pub enum SwapInstruction {
     ///   14. `[]` Token B program id
     WithdrawAllTokenTypes(WithdrawAllTokenTypes),
 
-    ///   Deposit one type of tokens into the pool.  The output is a "pool" token
-    ///   representing ownership into the pool. Input token is converted as if
-    ///   a swap and deposit all token types were performed.
+    ///   Deposit one type of tokens into the pool.  The output is a "pool"
+    /// token   representing ownership into the pool. Input token is
+    /// converted as if   a swap and deposit all token types were performed.
```
Comment (joncinque) [inline token-swap/program/src/instruction.rs:186]: spacing off here

File: token/cli/src/main.rs [hunk 32]
```diff
@@ -3378,8 +3387,8 @@ async fn command_deposit_withdraw_confidential_tokens(
         // // TODO: expose account balance decryption in token
         // let aes_key = aes_key.expect("AES key must be provided");
         // let current_balance = token
-        //     .confidential_transfer_get_available_balance_with_key(&token_account_address, aes_key)
-        //     .await?;
+        //     .confidential_transfer_get_available_balance_with_key(&
+        // token_account_address, aes_key)     .await?;
```
Comment (joncinque) [inline token/cli/src/main.rs:3391]: probably fine to leave as is, we'll fix it later

File: token/program-2022/src/extension/confidential_transfer/instruction.rs [hunk 33]
```diff
@@ -350,55 +372,65 @@ pub enum ConfidentialTransferInstruction {
...
+    ///   5. `[]` Context state account for
+    ///      `VerifyBatchedGroupedCiphertext2HandlesValidityProof`.
     ///   6. `[]` Context state account for `VerifyBatchedRangeProofU128`.
     ///   7. `[signer]` The source account owner.
-    ///   If `close_split_context_state_on_execution` is set, all context state accounts must be
-    ///   `writable` and the following additional sequence of accounts are needed:
-    ///   8. `[]` The destination account for lamports from the context state accounts.
+    ///   If `close_split_context_state_on_execution` is set, all context state
+    /// accounts must be   `writable` and the following additional sequence
+    /// of accounts are needed:
```
Comment (joncinque) [inline token/program-2022/src/extension/confidential_transfer/instruction.rs:405]: these should be indented / spacing fixed

File: token/program-2022/src/extension/confidential_transfer/instruction.rs [hunk 34]
```diff
@@ -350,55 +372,65 @@ pub enum ConfidentialTransferInstruction {
...
+    ///   7. `[]` Context state account for
+    ///      `VerifyBatchedGroupedCiphertext2HandlesValidityProof`.
     ///   8. `[]` Context state account for `VerifyBatchedRangeProofU256`.
     ///   9. `[signer]` The source account owner.
-    ///   If `close_split_context_state_on_execution` is set, all context state accounts must be
-    ///   `writable` and the following additional sequence of accounts are needed:
-    ///   10. `[]` The destination account for lamports from the context state accounts.
+    ///   If `close_split_context_state_on_execution` is set, all context state
+    /// accounts must be   `writable` and the following additional sequence
+    /// of accounts are needed:
```
Comment (joncinque) [inline token/program-2022/src/extension/confidential_transfer/instruction.rs:426]: this should be indented / spacing fixed

**Review by joncinque — COMMENTED** [review]
There's definitely some rough edges, but I think we'll eventually be able to land this! There were a lot of places that created weirdness, and sometimes my review comments are a bit terse, so please don't take them negatively. Once we cleanup those bits, this should be good to go. Please do it in new commits so that I don't have to go through the whole thing again :pray:

**Comment by buffalojoec** [comment]
> Once we cleanup those bits, this should be good to go.

😅 😅 😅 

I will probably not do this right away hah.

Thanks for a thorough review though!

**Comment by joncinque** [comment]
> I will probably not do this right away hah.

Please do it on the sooner side to avoid merge conflicts later! I really don't want to go through all of this again.

**Comment by buffalojoec** [comment]
> Please do it on the sooner side to avoid merge conflicts later!

Ah nuts.

File: token/transfer-hook/example/src/processor.rs (hunk 15 above)
Comment (buffalojoec) [reply token/transfer-hook/example/src/processor.rs:78]: Ugly, but ok!

File: token/cli/src/main.rs (hunk 32 above)
Comment (buffalojoec) [reply token/cli/src/main.rs:3391]: Fixed it a little.

File: binary-oracle-pair/program/src/instruction.rs [hunk 35]
```diff
@@ -55,12 +56,13 @@ pub enum PoolInstruction {
     Deposit(u64),
 
     ///   Withdraw from the pool.
-    ///   If current slot is < mint_end slot, 1 Pass AND 1 Fail token convert to 1 deposit
-    ///   If current slot is > decide_end_slot slot && decide == Some(true), 1 Pass convert to 1 deposit
-    ///   otherwise 1 Fail converts to 1 deposit
+    ///   If current slot is < mint_end slot, 1 Pass AND 1 Fail token convert to
+    /// 1 deposit   If current slot is > decide_end_slot slot && decide ==
```
[the remark this answers is not in the document]
Comment (buffalojoec) [reply binary-oracle-pair/program/src/instruction.rs:60]: What the hell is this program?

File: governance/program/src/error.rs [hunk 36]
```diff
@@ -15,7 +15,8 @@ use {
 pub enum GovernanceError {
     /// Invalid instruction passed to program
     #[error("Invalid instruction passed to program")]
-    InvalidInstruction = 500, // Start Governance custom errors from 500 to avoid conflicts with programs invoked via CPI
+    InvalidInstruction = 500, /* Start Governance custom errors from 500 to avoid conflicts
+                               * with programs invoked via CPI */
```
[the remark this answers is not in the document]
Comment (buffalojoec) [reply governance/program/src/error.rs:19]: I _think_ double-slash comments are ignored by thiserror

**Comment by buffalojoec** [comment]
Holy schmoly,

done

File: binary-oracle-pair/program/src/instruction.rs (hunk 35 above)
[the remark this answers is not in the document]
Comment (joncinque) [reply binary-oracle-pair/program/src/instruction.rs]: I think it was a Jarry contribution. I can't remember off the top of my head what it's for, and I'm too lazy to read the code haha

File: governance/program/src/error.rs [hunk 37]
```diff
@@ -11,11 +11,13 @@ use {
 };
 
 /// Errors that may be returned by the Governance program
+// Start Governance custom errors from 500 to avoid conflicts with programs
+// invoked via CPI
```
Comment (joncinque) [inline governance/program/src/error.rs:15]: Much better!

File: governance/program/src/instruction.rs [hunk 38]
```diff
@@ -142,45 +168,60 @@ pub enum GovernanceInstruction {
...
+    ///   5. `[]` Governing TokenOwnerRecord account (Used only if not signed by
+    ///      RealmAuthority)
     ///   6. `[signer]` Payer
     ///   7. `[]` bpf_upgradeable_loader program
     ///   8. `[]` System program
     ///   9. `[]` Sysvar Rent
     ///   10. `[signer]` Governance authority
-    ///   11. `[]` RealmConfig account. PDA seeds: ['realm-config', realm]
+    ///   11. `[]` RealmConfig account.
+    ///     * PDA seeds: ['realm-config', realm]
```
Comment (joncinque) [inline governance/program/src/instruction.rs:186]: I really like these!

**Review by joncinque — DISMISSED** [review]
Great job! You deserve the biggest salute

salute

**Comment by buffalojoec** [comment]
### Merge activity

* **Nov 8, 3:35 PM**: @buffalojoec started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/solana-labs/solana-program-library/5744).

**Pull request merged**